### PR TITLE
feat(studio): add gated Fal H3 Max Turbo motion generation

### LINKED ISSUE #380: feat(studio): A→B pose interpolation and single-still action generation through fal H3 Max Turbo, inside Studio
## Summary

Two generation actions that live **inside Studio**, not in the separate Workflow canvas:

1. **A→B interpolation.** Pose the character at frame A, pose it at frame B, press one button. Studio renders both stills, the server sends them to fal `minimax/h3-max-turbo/image-to-video` as `image_url` + `end_image_url`, and the clip comes back into the scene.
2. **Single-still action.** Pose once, tell the Studio Agent what the character should do ("swing the sword overhead", "sit on the box"). The Agent writes the motion prompt, the server sends the one still as `image_url`, and the clip comes back.

Both go through the same server proxy on `api.cozyclay.org`. The browser never holds the fal key.

This is the launch feature for the one-week free promo (10 clips per user per day, then paid).

## Why fal, why now

- Tested every other route for this: local H3 on a PRO 6000 (4-step + SLA, ~20 s per 5 s clip), ComfyUI turbo LoRAs, Artlist. fal Turbo is the only one fast and cheap enough to sit behind a button.
- Measured today on the real endpoint (480P, 5 s, A→B): **inference 0.46–0.52 s, submit-to-result 2.6–2.9 s wall clock, $0.0625 per clip** at the launch rate. Standard rate after Sept 30 is $0.125.
- fal is also the licensed route for end users. Self-hosting H3 weights for users in excluded territories is not an option.

## User flow: A→B in Studio

1. User poses the character (existing Pose Studio) and marks it as **A**.
2. User poses again and marks it as **B**. Camera and framing stay locked between the two.
3. **Interpolate** button. Studio renders A and B as stills using the current camera (same render path as the mocap-contract still: frontal, full body, plain background).
4. Client POSTs both stills to `api.cozyclay.org`. Server uploads to fal storage, submits, polls, returns the video URL and the real width/height/fps/duration.
5. Clip appears in the scene as a motion source, with A and B still attached so the user can regenerate with a different prompt or seed.
6. From here it feeds the existing extract path (#370 covers the GVHMR bridge side).

## User flow: single still + action via Studio Agent

1. User poses once.
2. In the Studio Agent composer: "make him draw the sword and step forward". The Agent turns that into the H3 motion prompt using the existing mocap prompt rules (fixed camera, full body visible, no cuts, motion beats in prose).
3. Same proxy, same endpoint, only `image_url`, no end frame.
4. Result returns to the scene the same way.

The Agent already exists (#365, #366). This adds one tool to it.

## Where it lives in Studio

- Interpolate button next to the pose controls, visible when two marked poses exist.
- Generation state (queued / running / done / failed) in the same place as the other async work in Studio, not a modal that blocks the scene.
- Result modal reuses `result-modal.jsx` with the clip, cost of that clip, and remaining daily quota.
- No new LiteGraph canvas. The `feat/cozyclay-gen-node` branch stays unmerged; this supersedes it for the Studio path.

## Server proxy (api.cozyclay.org)

Endpoint sketch:

```
POST /v1/motion/interpolate     { stillA, stillB, prompt?, duration=5 }
POST /v1/motion/act             { still, prompt, duration=5 }
GET  /v1/motion/jobs/:id        → { status, video, width, height, fps, duration, cost }
```

Server side, in order:

1. Auth the user. Look up today's count. Reject with a clear error at 10.
2. Upload each still to fal storage:
   `POST https://rest.alpha.fal.ai/storage/upload/initiate?storage_type=fal-cdn-v3` → `upload_url`, `file_url`, then `PUT upload_url`.
3. `POST https://queue.fal.run/minimax/h3-max-turbo/image-to-video` with
   `{ image_url, end_image_url?, prompt, duration: 5, resolution: "480P", prompt_expansion_mode: "disabled" }`
4. Poll `status_url`, fetch `response_url`, return `video.url` plus `timings.inference`.
5. Increment the user's daily count only after a successful result.
6. `FAL_KEY` lives in the server environment. Scope: API (inference only, cannot touch billing).

### Hard requirement: hosted URLs, not base64

Passing the stills as `data:image/jpeg;base64,...` in `image_url` **fails silently**. The request succeeds, but the model treats it as text-to-video and invents a character. Verified today: same prompt, base64 → random superhero; fal storage URL → our humanoid, correct A→B motion. Always upload first.

## Quota and promo

- Free week: 10 clips per user per day, server-enforced, reset at 00:00 UTC.
- After the week: paid per clip. Cost basis at standard rate is $0.125 per 480P 5 s clip; price it at 3–4x.
- fal account has auto top-up on ($10 when balance drops below $5) and a low-balance email at $10. Raise both before launch.
- If the promo runs before Sept 30, every clip is half price. That is the single biggest cost lever.

## Constraints

- Duration is an integer, 5–15 s. Default 5. Longer clips cost proportionally more and the interpolation does not get better with length.
- 480P output is 832×480 for a 16:9-ish input. Keep it. It is what the extract path expects and it is the cheapest tier.
- Turbo has no reference-to-video mode. Character identity comes only from the stills, so the still must be clean: frontal, full body, plain background. The existing mocap-contract render already does this.
- Turbo is fal's post-trained + distilled H3. Stylized inputs can drift toward "prettier" renders. Watch for it in QA; the test today held identity fine.

## Out of scope for this issue

- Camera-controls generation (`minimax/h3-max/camera-controls`). Separate feature, separate issue after launch.
- Reference-to-video / character sheets.
- Payment integration. This issue ends at "quota enforced, cost per clip known".
- GVHMR extract wiring: #370.

## Acceptance

- [ ] Two marked poses → one click → clip in the scene, under 5 s wall clock at 480P/5 s.
- [ ] One pose + Agent instruction → clip in the scene through the same proxy.
- [ ] Browser never sees `FAL_KEY`. All fal traffic originates from `api.cozyclay.org`.
- [ ] Stills reach fal as hosted URLs. Base64 path removed or rejected.
- [ ] Daily quota enforced server-side and shown in the result modal.
- [ ] Returned clip carries real width/height/fps/duration, not assumed values.
- [ ] Cost per clip logged per job so the promo spend is visible day by day.

## Related

- #370 fal request alignment with the mocap contract and GVHMR bridge
- #365 #366 Studio Agent composer and session persistence
- Branch `feat/cozyclay-gen-node`: superseded for the Studio path

### PR DESCRIPTION
**Pull request by HaD0Yun**
## What changed

Closes #380. Adds a Studio-only Fal H3 Max Turbo motion path for A→B pose interpolation and single-still action generation through Studio Agent.

- Captures references through the existing shot-camera render path at fixed 16:9 1920×1080; H3 480P returns 832×480-compatible video.
- Requires shaded body-part palette segmentation for every A/B reference and rejects B when camera position, rotation, or FOV moved after A.
- Uploads stills to Fal storage, submits `minimax/h3-max-turbo/image-to-video`, polls the queue, and returns the real video metadata.
- Sends hosted image URLs, `resolution: 480P`, and `prompt_expansion_mode: disabled`; prompts include fixed camera, full-body framing, and no cuts/reframing.
- Sends Fal REST input fields at the top level. The previous `{ input: ... }` wrapper was accepted by the queue but produced a result-stage 422; this is covered by the updated API test.
- Adds D1 job/usage records, success-only quota accounting (10 clips/day), request IDs, cost ledger fields, and account-scoped polling.
- Attaches the completed video to the existing motion ingest/extraction path and shows it in the result modal with metadata.
- Keeps `MOTION_GENERATION_ENABLED=false` in production and development until owner QA is complete.

## Validation

- `npm test` — PASS 199 Node verification files
- `npm run build` — PASS
- `node test/demo/verify-motion-api.mjs` — PASS (hosted upload, top-level Fal REST input, 480P, prompt expansion disabled, polling, metadata, QA lock)
- Manual Chrome QA on `/app/`: Flat body-part colours enabled; Mark A and Mark B each captured 1920×1080 refs. The UI now shows explicit green A/B `Captured` cards, Flat part counts, labelled thumbnails, and `A/B camera framing matched`; A→B and action generation remain disabled under the QA lock.
- Live Fal direct REST A→B test — COMPLETED, request `01a0b48a-3e9a-7050-a24a-989212ec5fb1`, inference `0.491s`, output `832×480`, `5.184s`.
- Live CozyClay `generateMotion()` path after the REST-body fix — COMPLETED, request `01a0b48d-d666-7a41-b4bc-ac8deb2df34d`, output `832×480`, `5.184s`; sampled frames preserved the A→B motion, flat body-part colours, full body, and fixed background/camera framing.

Fal model contract reference: https://fal.ai/models/minimax/h3-max-turbo/image-to-video/api

## UX follow-up

- Adds explicit A→B and A-only mode tabs with a four-step progress indicator (Flat → A → B → Generate).
- Adds a direct Flat segmentation CTA, Korean capture guidance, per-reference status/counts, recapture and B removal actions, and larger reference thumbnails.
- Locks camera preset/FOV and viewport pointer input after A capture; exposes unlock, drift detection, and restore-to-A-camera recovery.
- Keeps generation QA-locked while making every disabled state explain its next action inline.
- Browser QA verified the real `/app/` flow: Flat ON, A captured at `1920×1080` with `Flat 14개`, camera controls disabled, and A-only prompt mode visible.

## Bad motion: what was measured (2026-09-21)

The retargeted take from the first A→B run spun ~150° a few frames in, on a clip whose character never turned. Measured on the extractor's own inputs:

- **fal clip was fine.** Static camera, no turn (frame-to-frame luma diff 0.4–1.0 over frames 0–12).
- **ViTPose swapped left/right at frame 6** on the shaded mannequin's rear three-quarter view: shoulder `L−R dx` −70 px for frames 0–5, +70 px from frame 7, at confidence 0.6–0.8. The mannequin has no face; from behind the 2D pose network mirrors it.
- **GVHMR turned the swap into a body spin**: `smpl_global_orient` yaw 6° → 55° (f6) → 116° (f7) → 153° (f120). Same in the retargeted cskel27 Hips.
- **The feet sat on the frame edge** (default camera, rig min-Y at −0.87 NDC; fal output bbox bottom = 480). Palette ankle confidence fell to 0.0–0.3 and the extractor lost the walk (root travel 0.26 m against 0.7 body-heights of travel in the video).
- The quality gate passed both spun takes: it measures foot slide / jitter / floor, not heading. Follow-up.

**Tried and dropped: forcing palette keypoints for fal clips.** On the same clip it held heading within 12° (no spin) but placed the right shoulder on the right shin — `rightUpperArm` #00D0FF and `rightShin` #00FFF5 are 27° apart in hue and the H3 render blends them; measured shoulder y = 175 / **349** px against hip y = 237. GVHMR then bent the spine 65° to fit (ViTPose: 5°). Palette hue spacing is a separate issue; that commit is not on this branch.

## Fix on this branch: auto-framed A/B captures (`fe9877c`)

Capture A parks the shot camera on a `falReference` preset before rendering: front three-quarter **relative to the subject's heading**, 38 mm, feet ~0.2 of the frame above the bottom edge. Capture B reuses A's recorded framing verbatim, so the camera lock holds and the two stills differ only in pose. The user no longer has to frame the reference by hand.

Regeneration from auto-framed refs (fal `01a0c1eb-4396`, 832×480, 124 f): feet clear (bbox bottom 393 / 480, ankle conf 0.7), body upright (spine tilt 5°), yaw span 71° vs 146° from the rear view. Heading still wobbles on ViTPose — the front ¾ view reduces it, it does not remove it.

Also measured: **A→B interpolation does not generate locomotion.** Two standing refs 1.2 m apart produced a clip where the character slides sideways with knees at 5–8°; GVHMR's static-foot prior correctly holds the root still. An explicit "walks four steps" prompt (`01a0c189-5268`) produced real walking that the extractor read back (root 1.43 m, knees 85–94°). Travel needs the A-only action mode with a walking instruction; interpolate is for pose transitions.

Browser QA (headless Chrome, worktree stack 5190/5191, GVHMR on ubuntu-baremetal): pressing Capture A moved the shot camera from (0.97, 1.62, 2.39) fov 45 to (4.82, 1.70, 5.75) fov 29.8 with the rig at −0.48…+0.44 NDC; Capture B after moving the character 1.2 m kept the same camera and reported `A/B 카메라 프레이밍 일치 확인 · 카메라 잠금`. Extraction requests go to `/ardy/extract` with the bridge default keypoints.

Validation: `npm run build` PASS, `node tools/run-tests.mjs` PASS (206 files; `verify-camera-move` covers the facing-relative preset and feet clearance).

\n\n

## Quality pass (#380, 2026-09-21)

Measured on the A-only bat-swagger clip (v13c, 243 f @ 24 fps) with `node tools/ardy/mocap-metrics.mjs`:

| take | jitter mm/f² | foot-slide cm/s | notes |
|---|---|---|---|
| v13c, sigma 1.2 (before) | 4.61 | 30.12 | stance ankle moonwalks 45 cm/s vs hips 49 |
| v13c, sigma 3 + foot anchor (real bridge) | **4.13** | **8.04** | same clip, new extract path |
| v8 walk-to-sit, before → after | 1.08 → 0.89 | 3.3 → 2.1 | regression clip |

Levers, each RED→GREEN:
- `buildH3MotionPrompt`: camera + scene + single-shot locks stay, the heading lock goes (v13b with all locks off: fal zoomed in, feet cropped). No tempo words. `test/verify-fal-motion-prompt.mjs`.
- Face marks → two eyes 6 units apart (`src/facing-marks.js`). fal's expansion read the old band as "a dark horizontal eye slit"; ViTPose L/R eye flips 181/243 → 69/243. `test/verify-facing-marks.mjs`.
- Runner `--smooth-sigma 3` on both extraction paths (sweep: none 13.6 / 1.2 4.76 / 2.0 4.03 / 3.0 3.42). `test/verify-gvhmr-detector-flag.mjs`.
- `stabilizeMotion({ anchorFeet })`: foot-anchored root re-integration, on by default in `extract.mjs`. GVHMR contact logits are unusable on rendered clips (r ≤ 0.09 with planted frames), so stance is derived from the take. `test/verify-motion-stabilize.mjs`.

Not taken: 768P (fal fills the frame at 87 % body height and the 8 GB box OOMs); prompt_expansion quality vs disabled (identical framing at the same seed).
- Hybrid 2D keypoints (runner `--keypoints hybrid`, auto on palette clips): where the render lost a limb part (a prop repainting the hand), the joint comes from the surviving part's axis instead of ViTPose. v14 bat clip: right wrist above shoulder 22 % → 85 % (2D); 3D jitter 11.4 → 9.1, slide 35 → 25.
- Hybrid redefined as palette body + ViTPose face (2D body jitter 29 → 8 px/f²; palette-only face collapsed travel 6.3 → 2.9 m). v15: head upright, arm at shoulder, travel 7.4 m ≈ video.

### CODE REVIEW

**Comment by HaD0Yun** [comment]
검증 업데이트:

- `npm run build` 통과
- `node test/verify-reference-slots.mjs` 통과
- `node test/verify-video-frames.mjs` 통과
- 브라우저 QA (`http://127.0.0.1:5280/app/`): 기본 Medium 프레임에서 전신 누락 오류가 표시되고 캡처가 거부됨; Camera → Wide 프리셋 적용 후 A/B 모두 `1920×1080 · Shaded 14개`로 캡처 성공
- 변경 커밋: `67defdc`
